feat: add eslint

ZEROM22 opened the pull request.

vanyauhalin commented on makefile.js:
```diff
@@ -282,4 +282,19 @@ app
     stdout.write(pack.version)
   })
 
+app.command("lint")
+  .describe("Run ESLint on all project files")
+  .option("--fix")
+  .action(async (options) => {
+    try {
+      const { fix } = options
+      let eslintCommand = "npx eslint ."
+      if (fix) eslintCommand = "npx eslint . --fix"
+      exec(eslintCommand)
+    } catch {
+      error("ESLint failed. Please fix the linting errors.")
+      exit(1)
+    }
+  })
+
```
You don't have to wrap the ESLint commands in the Makefile. Simply add `eslint` to the `package.json` scripts:

```sh
$ cat ./package.json | jq .scripts
{
  "lint": "eslint ."
}
```

Then execute it:

```sh
$ pnpm lint
# or
$ pnpm lint --fix
```

vanyauhalin commented on .eslintrc.json:
Their comments don't reflect your personal opinion or provide any additional information. They simply repeat everything that can be found in the ESLint's documentation. I recommend deleting them.

vanyauhalin commented:
Additionally, please rename the `.eslintrc.json` to `eslint.config.js`. Also, add the `@stylistic/eslint-plugin-js` as a dependency.

Read more about the file [here](https://eslint.org/docs/latest/use/configure/configuration-files-new), and about the dependency [here](https://eslint.style/guide/why).

vanyauhalin commented:
> Additionally, please rename the `.eslintrc.json` to `eslint.config.js`. Also, add the `@stylistic/eslint-plugin-js` as a dependency.
> 
> Read more about the file [here](https://eslint.org/docs/latest/use/configure/configuration-files-new), and about the dependency [here](https://eslint.style/guide/why).

And please migrate to the flat config. Read more [here](https://eslint.org/docs/latest/extend/plugin-migration-flat-config).

ZEROM22 commented:
> Additionally, please rename the `.eslintrc.json` to `eslint.config.js`. Also, add the `@stylistic/eslint-plugin-js` as a dependency.
> 
> Read more about the file [here](https://eslint.org/docs/latest/use/configure/configuration-files-new), and about the dependency [here](https://eslint.style/guide/why).

I added `@stylistic/eslint-plugin-js' and rewrote the config to flat.

vanyauhalin commented:
@ZEROM22, please resolve conflicts.
